Woodoku/src/Entity/WoodokuBoard.py

```diff
@@ -22,86 +25,167 @@ def remove_blocks(self, blocks_coord: List[Tuple[int, int]]) -> None:
         pass
 
     def is_occupied(self, blocks_coord: List[Tuple[int, int]]) -> bool:
-        """
-        Check if each coord has block on it. If all of those coords are occupied, return true. Otherwise, false.
+        """Check if each coord has a block on it. If all of those coords are
+        occupied, return true. Otherwise, false.
+
+        Args:
+            blocks_coord (List[Tuple[int, int]]): list of coordinates to check
+
+        Raises:
+            CannotPlaceShapeError: raised when checking a invalid block
 
-        :param blocks_coord:
-        :return:
+        Returns:
+            bool: if any coordinates in `blocks_coord` is occupied
         """
-        pass
+        # TODO: raise the Error, according to docstring, the code below is just
+        # a placeholder
+        raise CannotPlaceShapeError(0, 0)
 
 
 class WoodokuBoard:
+    """A 9x9 Woodoku Board"""
+
     __scoreAgent: ScoreAgent
-    __representation: WoodokuBoardRepresentation
+    __representation: _WoodokuBoardRepresentation
 
     def __init__(self):
-        pass
-
-    def initialize(self):
-        pass
+        self.__scoreAgent = ScoreAgent()
+        self.__representation = _WoodokuBoardRepresentation()
 
     def can_add_shape_to_board(self, shape: WoodokuShape) -> bool:
+        """Check if the woodoku shape can fit into the board. If all current
+        shapes are not be able to add, the game fails.
+
+        Exhaustively scans the board in a random fashion to average the complexity
+
+        Args:
+            shape (WoodokuShape): a woodoku shape for validation
+
+        Returns:
+            bool: if the `shape` fits the board
         """
-        Check if the woodoku shape can fit into the board. If all current shapes are not be able to add, the game fails.
-        :return: if can fit.
-        """
-        pass
+        # TODO: improve performance
```

Comment:
Might want to delete this comment. Instead, create a new issue with detailed explanation.

TBH, performance on the game is not the highest priority.